### PR TITLE
Inform user of fallback to HTTP

### LINKED ISSUE #573: Inform user of fallback to HTTP.
Regards https://github.com/GoogleContainerTools/jib/issues/545#issuecomment-404378398

### PR DESCRIPTION
**Pull request by TadCordle**
Fixes #573

### CODE REVIEW

File: jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetrieverIntegrationTest.java [hunk 1]
```diff
@@ -24,17 +25,20 @@
 /** Integration tests for {@link AuthenticationMethodRetriever}. */
 public class AuthenticationMethodRetrieverIntegrationTest {
 
+  private static final EmptyBuildLogger BUILD_LOGGER = new EmptyBuildLogger();
```
Comment (briandealwis) [inline jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetrieverIntegrationTest.java]: Can't this just be a mock?

Comment (TadCordle) [reply jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/AuthenticationMethodRetrieverIntegrationTest.java]: Wasn't sure if it was a good idea to use mocks in integration tests.

File: jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/BlobCheckerIntegrationTest.java [hunk 2]
```diff
@@ -28,11 +29,12 @@
 public class BlobCheckerIntegrationTest {
 
   @ClassRule public static LocalRegistry localRegistry = new LocalRegistry(5000);
+  private final EmptyBuildLogger buildLogger = new EmptyBuildLogger();
```
Comment (chanseokoh) [inline jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/BlobCheckerIntegrationTest.java]: Let's be consistent. `static final ... BUILD_LOGGER`?

**Comment by coollog** [comment]
Oh for the `BuildLogger`, we should probably not have the `http` package classes depend on `builder` package classes (since the `http` package is lower-level and should be decoupled from the Jib execution - `builder` should be the highest level package). We could either have the `http` package have its own logger, or have some way of reporting back log messages that doesn't require injecting a `builder.BuildLogger`, or move the `BuildLogger` to a lower-level package.

File: jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/BlobCheckerIntegrationTest.java (hunk 2 above)
Comment (briandealwis) [reply jib-core/src/integration-test/java/com/google/cloud/tools/jib/registry/BlobCheckerIntegrationTest.java]: Or even just a singleton: `EmptyBuilderLogger.INSTANCE`.

**Comment by TadCordle** [comment]
@coollog I think `BuildLogger` fits in the root package with `Timer`, WDYT?

**Comment by coollog** [comment]
@TadCordle We can probably have it there for now, though the `Timer` is mostly a debugging thing that should probably be replaced with a more robust solution. With #714, we'll probably be able to rip out the timer logic since eventing will allow clients to handle their own timing.

**Comment by briandealwis** [comment]
Should we rename `BuildLogger` to something like `JibLogger` or `JibReporting`?

**Comment by TadCordle** [comment]
> Should we rename BuildLogger to something like JibLogger or JibReporting?

I think I'll submit a followup for this, it changes a lot of files.

**Pull request merged**